pytest_mergify/flaky_detection.py

```diff
@@ -20,6 +21,30 @@
 _MIN_TEST_RETRY_BUDGET_DURATION = datetime.timedelta(seconds=4)
 
 
+class _NewTestMetrics:
+    "Represents metrics collected for a new test."
+
+    initial_duration: datetime.timedelta = datetime.timedelta()
+    "Represents the duration of the initial execution of the test."
+
+    retry_count: int = 0
+    "Represents the number of times the test has been retried so far."
+
+    scheduled_retry_count: int = 0
+    "Represents the number of retries that have been scheduled for this test depending on the budget."
+
+    total_duration: datetime.timedelta = datetime.timedelta()
+    "Represents the total duration spent executing this test, including retries."
+
+    def _add_duration(self, duration: datetime.timedelta) -> None:
+        self.retry_count += 1
+        self.total_duration += duration
+
+    def _add_initial_duration(self, duration: datetime.timedelta) -> None:
+        self.initial_duration = duration
+        self._add_duration(duration)
```

Comment:
Just moving some logic from `FlakyDetector` here, but it simplifies the inteface :) 

```suggestion
        if not self.initial_duration:
            self.initial_duration = duration
        self.total_duration += duration
```